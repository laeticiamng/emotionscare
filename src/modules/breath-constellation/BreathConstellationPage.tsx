"use client";
import React from "react";
import { PageHeader, Card, Button } from "@/COMPONENTS.reg";
import { ConstellationCanvas } from "@/COMPONENTS.reg";
import { useBreathPattern, Pattern } from "@/ui/hooks/useBreathPattern";
import { useSound } from "@/COMPONENTS.reg"; // si P5 dispo
import { ff } from "@/lib/flags/ff";
import { recordEvent } from "@/lib/scores/events"; // si P6 dispo

const PATTERNS: Record<string, Pattern> = {
  "coherence-5-5": [
    { phase: "inhale", sec: 5 },
    { phase: "exhale", sec: 5 },
  ],
  "4-7-8": [
    { phase: "inhale", sec: 4 },
    { phase: "hold", sec: 7 },
    { phase: "exhale", sec: 8 },
  ],
  "box-4-4-4-4": [
    { phase: "inhale", sec: 4 },
    { phase: "hold", sec: 4 },
    { phase: "exhale", sec: 4 },
    { phase: "hold2", sec: 4 },
  ],
  "triangle-4-6-8": [
    { phase: "inhale", sec: 4 },
    { phase: "hold", sec: 6 },
    { phase: "exhale", sec: 8 },
  ],
};

export default function BreathConstellationPage() {
  const [patternKey, setPatternKey] = React.useState<keyof typeof PATTERNS>("coherence-5-5");
  const [cycles, setCycles] = React.useState(8);
  const [density, setDensity] = React.useState(0.8); // 0..1
  const reduced = typeof window !== "undefined" && window.matchMedia?.("(prefers-reduced-motion: reduce)").matches;

  const bp = useBreathPattern(PATTERNS[patternKey], cycles);
  const useNewAudio = ff?.("new-audio-engine") ?? false;
  const cue = useNewAudio ? useSound?.("/audio/tick.mp3") ?? null : null;

  const lastPhase = React.useRef<string>("");
  React.useEffect(() => {
    const p = bp.current.phase;
    if (p !== lastPhase.current) {
      lastPhase.current = p;
      if (cue?.play) cue.play().catch(() => {});
      if ("vibrate" in navigator && !reduced) navigator.vibrate?.(6);
    }
  }, [bp.current.phase, cue, reduced]);

  React.useEffect(() => {
    if (!bp.running && bp.cycle >= cycles && cycles > 0) {
      const dur = PATTERNS[patternKey].reduce((s, p) => s + p.sec, 0) * cycles;
      recordEvent?.({
        module: "breath-constellation",
        startedAt: new Date(Date.now() - dur * 1000).toISOString(),
        endedAt: new Date().toISOString(),
        durationSec: dur,
        score: Math.min(10, Math.round(cycles * 1.2)),
        meta: { pattern: patternKey, density },
      });
    }
  }, [bp.running, bp.cycle, cycles, density, patternKey]);

  return (
    <main aria-label="Breath Constellation">
      <PageHeader title="Breath Constellation" subtitle="Respiration guidée par une constellation vivante" />
      <Card>
        <section style={{ display: "grid", gap: 12 }}>
          <div style={{ display: "grid", gap: 8 }}>
            <label>
              Pattern
              <select value={patternKey} onChange={(e) => setPatternKey(e.target.value as any)}>
                <option value="coherence-5-5">Cohérence (5-5)</option>
                <option value="4-7-8">4-7-8</option>
                <option value="box-4-4-4-4">Box (4-4-4-4)</option>
                <option value="triangle-4-6-8">Triangle (4-6-8)</option>
              </select>
            </label>

            <label>
              Cycles : {cycles}
              <input type="range" min={4} max={16} step={1} value={cycles} onChange={(e) => setCycles(parseInt(e.target.value, 10))} />
            </label>

            <label>
              Densité : {Math.round(density * 100)}%
              <input type="range" min={0.3} max={1} step={0.05} value={density} onChange={(e) => setDensity(parseFloat(e.target.value))} />
            </label>
          </div>

          <ConstellationCanvas phase={bp.current.phase} phaseProgress={bp.phaseProgress} reduced={!!reduced} density={density} />

          <div style={{ display: "flex", gap: 8 }}>
            <Button onClick={bp.toggle} data-ui="primary-cta">{bp.running ? "Pause" : "Démarrer"}</Button>
            {!bp.running && bp.cycle > 0 && bp.cycle < cycles && <Button onClick={bp.start}>Reprendre</Button>}
            <Button onClick={bp.stop}>Arrêter</Button>
          </div>

          <small>Cycle {Math.min(bp.cycle, cycles)} / {cycles}</small>
        </section>
      </Card>
    </main>
  );
}

