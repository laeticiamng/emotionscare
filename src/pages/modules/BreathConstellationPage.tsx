import React, { useState, useEffect, useCallback, useMemo } from 'react';
import { motion, AnimatePresence } from 'framer-motion';
import { UniverseEngine } from '@/components/universe/UniverseEngine';
import { RewardAnimation } from '@/components/rewards/RewardAnimation';
import { useCollectionStore } from '@/store/collection.store';
import { UNIVERSE_CONFIGS } from '@/data/universes/config';
import { Button } from '@/components/ui/button';
import { Sparkles, Circle, Star } from 'lucide-react';

type BreathPhase = 'inhale' | 'hold' | 'exhale' | 'pause';
type SessionState = 'setup' | 'breathing' | 'complete';

const BREATH_PRESETS = {
  cosmic: {
    label: 'Équilibre cosmique (4-4-6-2)',
    description: 'Rythme progressif utilisé par la constellation pour guider la respiration.',
    timings: { inhale: 4, hold: 4, exhale: 6, pause: 2 },
  },
  coherence: {
    label: 'Cohérence cardiaque (5-5)',
    description: 'Respiration cadencée pour activer la cohérence cardiaque.',
    timings: { inhale: 5, hold: 0, exhale: 5, pause: 0 },
  },
  '4-7-8': {
    label: 'Relaxation 4-7-8',
    description: 'Technique apaisante pour calmer le système nerveux.',
    timings: { inhale: 4, hold: 7, exhale: 8, pause: 0 },
  },
} as const;

type BreathPresetKey = keyof typeof BREATH_PRESETS;
type BreathPreset = (typeof BREATH_PRESETS)[BreathPresetKey];

const BREATH_PRESET_ENTRIES = Object.entries(BREATH_PRESETS) as Array<[
  BreathPresetKey,
  BreathPreset
]>;

const BreathConstellationPage = () => {
  const [sessionState, setSessionState] = useState<SessionState>('setup');
  const [breathPhase, setBreathPhase] = useState<BreathPhase>('inhale');
  const [cycleCount, setCycleCount] = useState(0);
  const [targetCycles, setTargetCycles] = useState(5);
  const [stars, setStars] = useState<Array<{ x: number; y: number; brightness: number; id: number }>>([]);
  const [showReward, setShowReward] = useState(false);
  const [isUniverseReady, setIsUniverseReady] = useState(false);
  const [sessionStartTime, setSessionStartTime] = useState<Date | null>(null);
  const [selectedPreset, setSelectedPreset] = useState<BreathPresetKey>('cosmic');

  const { addReward } = useCollectionStore();
  const universe = UNIVERSE_CONFIGS.vrBreath;

  // Breath timing configuration (in seconds)
  const breathTiming = useMemo(() => ({
    ...BREATH_PRESETS[selectedPreset].timings,
  }), [selectedPreset]);

  // Start breathing session
  const startBreathing = useCallback(() => {
    setSessionState('breathing');
    setSessionStartTime(new Date());
    setCycleCount(0);
    setStars([]);
    setBreathPhase('inhale');
  }, []);

  // Breath cycle management
  useEffect(() => {
    if (sessionState !== 'breathing') return;

    let timer: NodeJS.Timeout;
    
    const nextPhase = () => {
      setBreathPhase(current => {
        switch (current) {
          case 'inhale':
            return 'hold';
          case 'hold':
            return 'exhale';
          case 'exhale':
            return 'pause';
          case 'pause':
            // Complete cycle - add a star
            const newStar = {
              id: Date.now(),
              x: 20 + Math.random() * 60, // Keep stars in center area
              y: 20 + Math.random() * 60,
              brightness: 0.5 + Math.random() * 0.5
            };
            setStars(prev => [...prev, newStar]);
            
            const newCycleCount = cycleCount + 1;
            setCycleCount(newCycleCount);
            
            if (newCycleCount >= targetCycles) {
              // Session complete
              setTimeout(() => {
                setSessionState('complete');
                generateConstellation();
              }, 1000);
              return 'inhale';
            }
            
            return 'inhale';
          default:
            return 'inhale';
        }
      });
    };

    const currentTiming = breathTiming[breathPhase];
    timer = setTimeout(nextPhase, Math.max(currentTiming, 0) * 1000);

    return () => clearTimeout(timer);
  }, [breathPhase, sessionState, cycleCount, targetCycles, breathTiming]);

  const generateConstellation = useCallback(() => {
    if (!sessionStartTime) return;

    const sessionDuration = Date.now() - sessionStartTime.getTime();
    
    const constellationData = {
      id: `constellation-${Date.now()}`,
      starPattern: stars,
      cycleCount,
      sessionDuration,
      timestamp: new Date(),
      preset: selectedPreset,
      presetLabel: BREATH_PRESETS[selectedPreset].label,
    };

    addReward({
      type: 'constellation',
      name: `Constellation ${getCycleQuality(cycleCount)}`,
      description: `${cycleCount} cycles de respiration - ${Math.round(sessionDuration / 1000)}s`,
      moduleId: 'breath-constellation',
      rarity: cycleCount >= 10 ? 'epic' : cycleCount >= 7 ? 'rare' : 'common',
      visualData: {
        color: '#4ecdc4',
        icon: 'constellation',
        animation: 'twinkle'
      },
      metadata: constellationData
    });

    setShowReward(true);
  }, [stars, cycleCount, sessionStartTime, addReward, selectedPreset]);

  const getCycleQuality = (cycles: number): string => {
    if (cycles >= 10) return 'Majeure';
    if (cycles >= 7) return 'Céleste';
    if (cycles >= 5) return 'Étoilée';
    return 'Naissante';
  };

  const getPhaseInstructions = (): string => {
    switch (breathPhase) {
      case 'inhale': return 'Inspirez lentement...';
      case 'hold': return 'Retenez...';
      case 'exhale': return 'Expirez doucement...';
      case 'pause': return 'Pause...';
      default: return '';
    }
  };

  const getSphereScale = (): number => {
    switch (breathPhase) {
      case 'inhale': return 1.5;
      case 'hold': return 1.4;
      case 'exhale': return 0.8;
      case 'pause': return 0.9;
      default: return 1;
    }
  };

  const resetSession = () => {
    setSessionState('setup');
    setCycleCount(0);
    setStars([]);
    setShowReward(false);
    setSessionStartTime(null);
  };

  if (!universe) {
    return <div>Univers Breath Constellation non trouvé</div>;
  }

  return (
    <>
      <UniverseEngine
        universe={universe}
        isEntering={true}
        onEnterComplete={() => setIsUniverseReady(true)}
        className="min-h-screen"
      >
        <div className="container mx-auto px-4 py-8">
          <div className="max-w-4xl mx-auto">
            
            {/* Setup phase */}
            {sessionState === 'setup' && (
              <motion.div
                className="text-center space-y-8"
                initial={{ opacity: 0, y: 20 }}
                animate={{ opacity: 1, y: 0 }}
              >
                <div className="text-white space-y-4">
                  <h2 className="text-3xl font-light">Constellation Respiratoire</h2>
                  <p className="text-lg opacity-80">
                    Chaque cycle de respiration créera une étoile dans votre constellation personnelle
                  </p>
                </div>

                <div className="space-y-6">
                  <div className="space-y-3">
                    <p className="text-white font-medium">Rythme de respiration</p>
                    <div className="grid gap-3 sm:grid-cols-3">
                      {BREATH_PRESET_ENTRIES.map(([key, preset]) => (
                        <Button
                          key={key}
                          type="button"
                          variant={selectedPreset === key ? 'default' : 'outline'}
                          onClick={() => setSelectedPreset(key)}
                          className="h-auto py-3 text-left text-white border-white/30"
                        >
                          <span className="block text-sm font-semibold">{preset.label}</span>
                          <span className="block text-xs text-white/70">{preset.timings.inhale}s / {preset.timings.hold}s / {preset.timings.exhale}s / {preset.timings.pause}s</span>
                        </Button>
                      ))}
                    </div>
                    <p className="text-sm text-white/60 max-w-2xl mx-auto">
                      {BREATH_PRESETS[selectedPreset].description}
                    </p>
                  </div>

                  <div className="space-y-4">
                    <label className="text-white font-medium">
                      Nombre de cycles: {targetCycles}
                    </label>
                    <div className="flex justify-center gap-2">
                      {[3, 5, 7, 10].map(count => (
                        <Button
                          key={count}
                          variant={targetCycles === count ? "default" : "outline"}
                          onClick={() => setTargetCycles(count)}
                          className="text-white border-white/30"
                        >
                          {count}
                        </Button>
                      ))}
                    </div>
                  </div>

                  <div className="text-white/60 space-y-2">
                    <p>• Inspirez : {breathTiming.inhale}s</p>
                    <p>• Retenez : {breathTiming.hold > 0 ? `${breathTiming.hold}s` : '—'}</p>
                    <p>• Expirez : {breathTiming.exhale}s</p>
                    <p>• Pause : {breathTiming.pause > 0 ? `${breathTiming.pause}s` : '—'}</p>
                  </div>

                  <Button
                    onClick={startBreathing}
                    size="lg"
                    className="bg-gradient-to-r from-blue-500 to-purple-600 text-white"
                  >
                    Commencer la session
                  </Button>
                </div>
              </motion.div>
            )}

            {/* Breathing phase */}
            {sessionState === 'breathing' && (
              <motion.div
                className="text-center space-y-8"
                initial={{ opacity: 0 }}
                animate={{ opacity: 1 }}
              >
                <div className="text-white space-y-2">
                  <h3 className="text-xl font-light">{getPhaseInstructions()}</h3>
                  <p className="text-sm opacity-60">{BREATH_PRESETS[selectedPreset].label}</p>
                  <p className="text-sm opacity-60">Cycle {cycleCount + 1} / {targetCycles}</p>
                </div>

                {/* Breathing sphere */}
                <div className="relative flex items-center justify-center h-96">
                  <motion.div
                    className="w-48 h-48 rounded-full bg-gradient-to-br from-blue-400 to-purple-600 flex items-center justify-center"
                    animate={{ 
                      scale: getSphereScale(),
                    }}
                    transition={{
                      duration: Math.max(breathTiming[breathPhase], 0.2),
                      ease: "easeInOut"
                    }}
                  >
                    <Circle className="w-24 h-24 text-white opacity-80" />
                  </motion.div>

                  {/* Existing stars from previous cycles */}
                  {stars.map((star, index) => (
                    <motion.div
                      key={star.id}
                      className="absolute"
                      style={{
                        left: `${star.x}%`,
                        top: `${star.y}%`,
                      }}
                      initial={{ scale: 0, opacity: 0 }}
                      animate={{ 
                        scale: 1, 
                        opacity: star.brightness,
                        rotate: [0, 180, 360]
                      }}
                      transition={{ 
                        duration: 2,
                        repeat: Infinity,
                        repeatType: "loop"
                      }}
                    >
                      <Star className="w-6 h-6 text-yellow-300" />
                    </motion.div>
                  ))}

                  {/* Breath particles */}
                  {breathPhase === 'exhale' && (
                    <div className="absolute inset-0">
                      {Array.from({ length: 8 }, (_, i) => (
                        <motion.div
                          key={i}
                          className="absolute w-2 h-2 bg-white rounded-full opacity-60"
                          style={{
                            left: '50%',
                            top: '50%',
                          }}
                          animate={{
                            x: Math.cos(i * 45 * Math.PI / 180) * 100,
                            y: Math.sin(i * 45 * Math.PI / 180) * 100,
                            opacity: [0.6, 0],
                            scale: [1, 0]
                          }}
                          transition={{
                            duration: Math.max(breathTiming.exhale, 0.2),
                            ease: "easeOut"
                          }}
                        />
                      ))}
                    </div>
                  )}
                </div>

                {/* Progress indicator */}
                <div className="w-64 mx-auto">
                  <div className="h-2 bg-white/20 rounded-full overflow-hidden">
                    <motion.div
                      className="h-full bg-gradient-to-r from-blue-400 to-purple-600"
                      initial={{ width: 0 }}
                      animate={{ width: `${(cycleCount / targetCycles) * 100}%` }}
                      transition={{ duration: 0.5 }}
                    />
                  </div>
                  <p className="text-white/60 text-sm mt-2">
                    {cycleCount} étoiles créées
                  </p>
                </div>
              </motion.div>
            )}

            {/* Complete phase */}
            {sessionState === 'complete' && (
              <motion.div
                className="text-center space-y-8"
                initial={{ opacity: 0, scale: 0.8 }}
                animate={{ opacity: 1, scale: 1 }}
              >
                <div className="text-white space-y-4">
                  <h3 className="text-2xl font-light">
                    Constellation {getCycleQuality(cycleCount)} créée !
                  </h3>
                  <p className="text-lg opacity-80">
                    {cycleCount} étoiles brillent dans votre ciel personnel
                  </p>
                  <p className="text-sm text-white/60">
                    Rythme : {BREATH_PRESETS[selectedPreset].label}
                  </p>
                </div>

                {/* Final constellation display */}
                <div className="relative w-80 h-80 mx-auto">
                  <div className="absolute inset-0 bg-gradient-to-br from-indigo-900 to-purple-900 rounded-full opacity-30" />
                  
                  {stars.map((star, index) => (
                    <motion.div
                      key={star.id}
                      className="absolute"
                      style={{
                        left: `${star.x}%`,
                        top: `${star.y}%`,
                      }}
                      animate={{
                        opacity: [star.brightness * 0.7, star.brightness, star.brightness * 0.7],
                        scale: [0.8, 1.2, 0.8]
                      }}
                      transition={{
                        duration: 2 + Math.random(),
                        repeat: Infinity,
                        delay: index * 0.2
                      }}
                    >
                      <Sparkles className="w-4 h-4 text-yellow-300" />
                    </motion.div>
                  ))}

                  {/* Constellation lines */}
                  <svg className="absolute inset-0 w-full h-full">
                    {stars.slice(0, -1).map((star, index) => {
                      const nextStar = stars[index + 1];
                      return (
                        <motion.line
                          key={`line-${index}`}
                          x1={`${star.x}%`}
                          y1={`${star.y}%`}
                          x2={`${nextStar.x}%`}
                          y2={`${nextStar.y}%`}
                          stroke="rgba(255, 255, 255, 0.3)"
                          strokeWidth="1"
                          initial={{ pathLength: 0 }}
                          animate={{ pathLength: 1 }}
                          transition={{ duration: 0.5, delay: index * 0.2 }}
                        />
                      );
                    })}
                  </svg>
                </div>

                <div className="flex justify-center gap-4">
                  <Button 
                    onClick={resetSession} 
                    variant="outline" 
                    className="text-white border-white/30"
                  >
                    Nouvelle constellation
                  </Button>
                </div>
              </motion.div>
            )}
          </div>
        </div>
      </UniverseEngine>

      {/* Reward animation */}
      {showReward && (
        <RewardAnimation
          reward={{
            id: `constellation-${Date.now()}`,
            type: 'constellation',
            name: `Constellation ${getCycleQuality(cycleCount)}`,
            description: `${cycleCount} étoiles créées • ${BREATH_PRESETS[selectedPreset].label}`,
            moduleId: 'breath-constellation',
            rarity: cycleCount >= 10 ? 'epic' : cycleCount >= 7 ? 'rare' : 'common',
            visualData: {
              color: '#4ecdc4',
              icon: 'constellation',
              animation: 'twinkle'
            },
            earnedAt: new Date()
          }}
          isVisible={showReward}
          onComplete={() => setShowReward(false)}
        />
      )}
    </>
  );
};

export default BreathConstellationPage;
