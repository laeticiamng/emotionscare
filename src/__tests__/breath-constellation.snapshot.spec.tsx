import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import BreathConstellationPage from "@/modules/breath-constellation/BreathConstellationPage";

// mock canvas context for jsdom
Object.defineProperty(HTMLCanvasElement.prototype, "getContext", {
  value: () => ({
    scale: () => {},
    setTransform: () => {},
    clearRect: () => {},
    beginPath: () => {},
    arc: () => {},
    fill: () => {},
    moveTo: () => {},
    lineTo: () => {},
    stroke: () => {},
    save: () => {},
    restore: () => {},
    fillStyle: "",
    globalAlpha: 1,
    lineWidth: 1,
    strokeStyle: "",
  }),
});

class RO {
  observe() {}
  disconnect() {}
}
(globalThis as any).ResizeObserver = RO;

describe("BreathConstellationPage", () => {
  it("rend la page et le CTA Démarrer", () => {
    const { container, getByText } = render(<BreathConstellationPage />);
    expect(getByText(/Breath Constellation/i)).toBeTruthy();
    expect(getByText(/Démarrer/i)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
