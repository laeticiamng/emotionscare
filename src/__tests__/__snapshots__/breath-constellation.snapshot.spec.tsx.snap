// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BreathConstellationPage > rend la page et le CTA Démarrer 1`] = `
<div>
  <main
    aria-label="Breath Constellation"
  >
    <div
      class="relative overflow-hidden rounded-2xl p-6 mb-8 bg-gradient-to-r from-primary/10 via-primary/5 to-background"
      style="opacity: 0; transform: translateY(-20px);"
    >
      <div
        class="absolute inset-0 bg-gradient-to-r from-blue-50/50 via-purple-50/30 to-pink-50/20 dark:from-blue-900/10 dark:via-purple-900/5 dark:to-pink-900/5"
      />
      <div
        class="relative z-10"
      >
        <div
          class="flex items-center justify-between mb-4"
        >
          <div
            class="flex items-center gap-4"
          >
            <button
              class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed focus-enhanced border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground hover:shadow-md active:scale-95 h-8 px-3 text-xs rounded-full"
              type="button"
            >
              <svg
                class="lucide lucide-arrow-left h-4 w-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m12 19-7-7 7-7"
                />
                <path
                  d="M19 12H5"
                />
              </svg>
            </button>
            <div
              class="flex items-center gap-3"
            >
              <div>
                <div
                  class="flex items-center gap-3 mb-1"
                >
                  <h1
                    class="text-2xl font-bold bg-gradient-to-r from-foreground to-foreground/80 bg-clip-text"
                  >
                    Breath Constellation
                  </h1>
                </div>
                <p
                  class="text-muted-foreground"
                >
                  Respiration guidée par une constellation vivante
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex items-center gap-2"
        >
          <button
            class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed focus-enhanced hover:bg-accent hover:text-accent-foreground active:scale-95 h-8 rounded-md px-3 text-xs"
            type="button"
          >
            <svg
              class="lucide lucide-heart h-3 w-3 mr-1"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
              />
            </svg>
            Favoris
          </button>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed focus-enhanced hover:bg-accent hover:text-accent-foreground active:scale-95 h-8 rounded-md px-3 text-xs"
            type="button"
          >
            <svg
              class="lucide lucide-share2 h-3 w-3 mr-1"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="18"
                cy="5"
                r="3"
              />
              <circle
                cx="6"
                cy="12"
                r="3"
              />
              <circle
                cx="18"
                cy="19"
                r="3"
              />
              <line
                x1="8.59"
                x2="15.42"
                y1="13.51"
                y2="17.49"
              />
              <line
                x1="15.41"
                x2="8.59"
                y1="6.51"
                y2="10.49"
              />
            </svg>
            Partager
          </button>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed focus-enhanced hover:bg-accent hover:text-accent-foreground active:scale-95 h-8 rounded-md px-3 text-xs"
            type="button"
          >
            <svg
              class="lucide lucide-star h-3 w-3 mr-1"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
              />
            </svg>
            Noter
          </button>
        </div>
      </div>
    </div>
    <div
      class="rounded-xl border bg-card text-card-foreground shadow"
    >
      <section
        style="display: grid; gap: 12px;"
      >
        <div
          style="display: grid; gap: 8px;"
        >
          <label>
            Pattern
            <select>
              <option
                value="coherence-5-5"
              >
                Cohérence (5-5)
              </option>
              <option
                value="4-7-8"
              >
                4-7-8
              </option>
              <option
                value="box-4-4-4-4"
              >
                Box (4-4-4-4)
              </option>
              <option
                value="triangle-4-6-8"
              >
                Triangle (4-6-8)
              </option>
            </select>
          </label>
          <label>
            Cycles : 
            8
            <input
              max="16"
              min="4"
              step="1"
              type="range"
              value="8"
            />
          </label>
          <label>
            Densité : 
            80
            %
            <input
              max="1"
              min="0.3"
              step="0.05"
              type="range"
              value="0.8"
            />
          </label>
        </div>
        <div
          style="width: 100%; height: 300px; border-radius: 12px; overflow: hidden;"
        >
          <canvas
            aria-label="Constellation respiratoire"
            height="0"
            style="width: 100%; height: 100%; display: block;"
            width="0"
          />
        </div>
        <div
          style="display: flex; gap: 8px;"
        >
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed focus-enhanced bg-primary text-primary-foreground shadow hover:bg-primary/90 hover:shadow-md active:scale-95 h-9 px-4 py-2"
            data-ui="primary-cta"
            type="button"
          >
            Démarrer
          </button>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed focus-enhanced bg-primary text-primary-foreground shadow hover:bg-primary/90 hover:shadow-md active:scale-95 h-9 px-4 py-2"
            type="button"
          >
            Arrêter
          </button>
        </div>
        <small>
          Cycle 
          0
           / 
          8
        </small>
      </section>
    </div>
  </main>
</div>
`;
